src/QuadGK.jl

```diff
@@ -78,7 +78,11 @@ function evalrule(f, a,b, x,w,gw, nrm)
     Ig *= s
     E = nrm(Ik - Ig)
     if isnan(E) || isinf(E)
-        throw(DomainError())
+        @static if VERSION < v"0.7.0"
+            throw(DomainError())
+        else
+            throw(DomainError(E))
```

Comment:
`E` is the wrong argument here.   We are really supposed to pass the `x` that triggered the domain error.   In this case I would pass the midpoint `a + s`.   Also, `DomainError` takes a second argument, a message, which we might as well take advantage of, e.g. `DomainError(a+s, "integrand produced $E in the interval ($a, $b)")`